pycrostates/cluster/aahc.py

```diff
@@ -0,0 +1,408 @@
+
+        if normalize_input:
+            data = cluster.copy()
+
+        GEV = np.sum(data * cluster, axis=0)
+
+        assignment = np.arange(n_frame)
+
+        while cluster.shape[1] > n_clusters:
+
+            to_remove = np.argmin(GEV)
+            orphans = assignment == to_remove
+
+            old_cluster = cluster[:, to_remove].copy()
+
+            cluster = np.delete(cluster, to_remove, axis=1)
+            GEV = np.delete(GEV, to_remove, axis=0)
+            assignment[assignment > to_remove] = (
+                assignment[assignment > to_remove] - 1
+            )
+
+            fit = data[:, orphans].T @ cluster
+            if ignore_polarity:
+                fit = np.abs(fit)
+            new_assignment = np.argmax(fit, axis=1)
+            assignment[orphans] = new_assignment
+
+            cluster_to_update = np.unique(new_assignment)
+            for c in cluster_to_update:
+                members = assignment == c
+                if ignore_polarity:
+
+                    old_weight = len(new_assignment == c) / members.sum()
+
+                    sgn = np.sign(old_cluster @ cluster[:, c])
+
+                    v0 = old_weight * sgn * old_cluster + \
+                        (1. - old_weight) * cluster[:, c]
+
+                    v, _ = AAHCluster._first_principal_component(
+                        data[:, members], tol, v0
+                    )
+                    cluster[:, c] = v
+                else:
+                    cluster[:, c] = np.mean(data[:, members], axis=0)
+                    cluster[:, c] /= np.linalg.norm(
+                        cluster[:, c], axis=0, keepdims=True
+                    )
+                new_fit = cluster[:, slice(c, c + 1)] * data[:, members]
+                if ignore_polarity:
+                    new_fit = np.abs(new_fit)
+                GEV[c] = np.sum(new_fit)
+        return cluster.T, assignment
+
+    @staticmethod
+    @njit(cache=True)
+    def _first_principal_component(
+        X: NDArray[float],
+        tol: float,
+        v0: Optional[NDArray[float]] = None,
+        max_iter: int = 100,
+    ) -> Tuple[NDArray[float], float]:
+        """Compute first principal component.
+
+        Parameters
+        ----------
+        X : numpy.ndarray
+            Input data matrix with dimensions (channels x observations)
+        tol : float
+            Tolerance for convergence.
+        v0 : numpy.ndarray or None (default)
+            Initial estimate of the principal vector with dimensions
+            (channels, )
+            If None, a random initialization is used
+        max_iter : int
+            Maximum number of iterations to estiamte the first principal
+            component.
+
+        Returns
+        -------
+
+        tuple(v, eig)
+
+        v : numpy.ndarray
+            estimated principal component vector
+        eig : float
+            eigenvalue of the covariance matrix of X (channels x channels)
```

Comment:
```suggestion
        v : numpy.ndarray
            Estimated principal component vector.
        eig : float
            Eigenvalue of the covariance matrix of X (channels x channels)
```

I'm going to guess `pydocstyle` would complain otherwise.